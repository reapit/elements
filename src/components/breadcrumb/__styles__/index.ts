import { styled } from '@linaria/react'

export const ElBreadCrumbItem = styled.a`
  font-weight: var(--breadcrumb-item-font-weight);
  border-radius: var(--breadcrumb-item-border-radius);
  font-size: var(--breadcrumb-item-font-size);
  color: var(--breadcrumb-item-font-color);
  margin-right: var(--breadcrumb-item-spacing);
`

export const ElBreadCrumbContainer = styled.div`
  display: flex;
  width: 100%;
`
