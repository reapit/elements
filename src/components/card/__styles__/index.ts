import { styled } from '@linaria/react'
import { css } from '@linaria/core'
import { isTablet } from '../../../styles/media'
import { ElDeprecatedAvatar } from '../../deprecated-avatar'

export const elCardFocussed = css`
  background-color: var(--purple-050);
`

export const ElCardWrap = styled.div`
  padding: 1rem;
  border-radius: 0.25rem;
  box-shadow: 2px 4px 20px rgba(0, 0, 0, 0.03);
  background: var(--white);
  position: relative;

  ${isTablet} {
    padding: 1.25rem;
    box-shadow: 2px 4px 20px rgba(0, 0, 0, 0.05);
  }

  &.${elCardFocussed} {
    background-color: var(--purple-050);
  }
`

export const elCardSubHeadingWrapAvatar = css``

export const ElCardHeadingWrap = styled.div`
  display: flex;
  flex-direction: column;
  overflow: hidden;
  width: 100%;

  &.${elCardSubHeadingWrapAvatar} {
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-between;
  }
`

export const ElCardMainWrap = styled.div`
  display: flex;
  flex-wrap: nowrap;

  ${ElDeprecatedAvatar} {
    margin-right: 1rem;

    ${isTablet} {
      margin-right: 1.25rem;
    }
  }
`

export const ElCardHeading = styled.h5`
  display: -webkit-box;
  -webkit-line-clamp: 1;
  -webkit-box-orient: vertical;
  overflow: hidden;
  white-space: normal;
  margin-bottom: 0.25rem;
  width: 100%;
  font-size: var(--font-size-small);
`

export const ElCardSubHeading = styled.h6`
  color: var(--neutral-500);
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  font-size: var(--font-size-smallest);
`

export const elCardSubHeadingAdditionalExpanded = css``

export const ElCardSubHeadingAdditional = styled.h6`
  color: var(--neutral-500);
  font-size: var(--font-size-smallest);
  font-weight: var(--font-weight-medium);
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  display: flex;
  justify-content: space-between;
`

export const elCardBodyWrapExpanded = css``

export const ElCardBodyWrap = styled.div`
  width: 100%;
  color: var(--black);
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
  white-space: normal;
  height: 3rem;
  font-size: var(--font-size-smallest);
  transition: height 0.2s linear;
  transition: margin-bottom 0.2s linear;
  margin-top: 0.5rem;

  ${isTablet} {
    height: 3.25rem;
    font-size: var(--font-size-small);
  }
`

export const elMobileListToggle = css`
  top: 1.5rem;
  right: 0;
`

export const ElCardAvatarWrap = styled.div`
  display: flex;
  flex-direction: column;
  flex-shrink: 0;
  align-items: center;
  justify-content: center;
  background-color: var(--neutral-100);
  margin-right: 0.5rem;
  border-radius: 50%;
  width: 2.5rem;
  height: 2.5rem;

  img {
    max-width: 2.5rem;
    border-radius: 50%;
  }
`

export const ElCardImageWrap = styled(ElCardAvatarWrap)`
  border-radius: 0;
  width: 4.5rem;
  height: 3.25rem;

  img {
    max-width: 4.5rem;
    border-radius: 0;
  }
`

export const ElCardList = styled.div`
  display: flex;
`

export const elCardListMainWrapExpanded = css``

export const ElCardListMainWrap = styled.div`
  display: flex;
  flex-wrap: nowrap;
  position: relative;
  flex-direction: column;
`

export const ElCardListHeading = styled.h5`
  width: 100%;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  font-size: var(--font-size-small);
  margin-bottom: 0.25rem;
`

export const ElCardListSubHeading = styled.h6`
  font-size: var(--font-size-smallest);
  color: var(--neutral-500);
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  margin-bottom: 0.75rem;
`

export const elCardListItemExpanded = css`
  height: 2.5rem;
  margin-bottom: 0.5rem;

  &:last-of-type {
    margin-bottom: 0;
  }
`

export const ElCardListItem = styled.div`
  display: flex;
  height: 2rem;
  margin-bottom: 0.5rem;
  overflow: hidden;

  &:last-of-type {
    margin-bottom: 0;
  }
`

export const ElCardListItemTextWrap = styled.div`
  font-size: var(--font-size-smallest);
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  cursor: pointer;
`

export const ElCardListItemTextPrimary = styled.div`
  color: var(--black);
`

export const ElCardListItemTextSecondary = styled.div`
  font-size: var(--font-size-smallest);
  color: var(--intent-primary);
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
  overflow: hidden;
`

export const ElCardListIcon = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 2em;
  width: 2em;
  margin-right: 0.5rem;
  flex-shrink: 0;
`
