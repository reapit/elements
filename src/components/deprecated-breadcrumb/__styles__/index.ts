import { styled } from '@linaria/react'

/** @deprecated */
export const ElDeprecatedBreadCrumbItem = styled.a`
  font-weight: inherit;
  border-radius: 0.5rem;
  font-size: var(--font-size-small);
  color: var(--intent-default);
  margin-right: 0.375rem;
`

/** @deprecated */
export const ElDeprecatedBreadCrumbContainer = styled.div`
  display: flex;
  width: 100%;
`
