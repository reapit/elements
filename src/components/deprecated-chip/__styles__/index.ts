import { styled } from '@linaria/react'

const checked = `data:image/svg+xml;utf8,<svg width="12px" height="12px" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M12.3608 13.8296L18.0208 19.4896C18.2652 19.775 18.6489 19.8993 19.0142 19.8114C19.3795 19.7235 19.6647 19.4383 19.7526 19.073C19.8405 18.7077 19.7162 18.324 19.4308 18.0796L13.7708 12.4196L19.4308 6.75958C19.7162 6.51519 19.8405 6.13146 19.7526 5.76616C19.6647 5.40086 19.3795 5.11564 19.0142 5.02777C18.6489 4.93989 18.2652 5.0642 18.0208 5.34958L12.3608 11.0096L6.70079 5.34958C6.31076 4.96185 5.68083 4.96185 5.29079 5.34958C4.90307 5.73961 4.90307 6.36954 5.29079 6.75958L10.9508 12.4196L5.29079 18.0796C4.90307 18.4696 4.90307 19.0995 5.29079 19.4896C5.68083 19.8773 6.31076 19.8773 6.70079 19.4896L12.3608 13.8296Z" fill="slategray"/>
</svg>
`

/** @deprecated */
export const ElDeprecatedChipCheckbox = styled.input`
  height: 0;
  width: 0;
  visibility: hidden;
  position: absolute;

  &:checked + label,
  &:hover + label {
    background: var(--purple-050);
    padding: 0.2rem 2rem 0.2rem 1rem;
    color: var(--black);

    &::before {
      content: '';
      position: absolute;
      background-image: url('${checked}');
      background-position: center center;
      background-repeat: no-repeat;
      height: 1rem;
      width: 1rem;
      right: 0.5rem;
      top: 0.25rem;
    }
  }
`

/** @deprecated */
export const ElDeprecatedChipLabel = styled.label`
  cursor: pointer;
  width: auto;
  height: 28px;
  background: var(--purple-050);
  border: 1px solid var(--purple-050);
  border-radius: 1rem;
  padding: 0.2rem 1.5rem;
  position: relative;
  font-size: var(--font-size-small);
  transition: all 0.2s linear;
  display: flex;
  align-items: center;
  justify-content: center;
  color: var(--neutral-500);

  &:hover {
    color: var(--neutral-700);
  }
`

/** @deprecated */
export const ElDeprecatedChipGroup = styled.div`
  display: grid;
`

/** @deprecated */
export const ElDeprecatedChipGroupInner = styled.div`
  display: flex;
  flex-wrap: wrap;
  grid-auto-flow: column;
  gap: 0.25rem;
  width: fit-content;
  height: fit-content;
`
