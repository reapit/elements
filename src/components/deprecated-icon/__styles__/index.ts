import { styled } from '@linaria/react'
import {
  elIntentPrimary,
  elIntentSuccess,
  elIntentDanger,
  elIntentNeutral,
  elIntentPending,
  elIntentWarning,
  elIntentDefault,
} from '../../../styles/intent'

/** @deprecated */
export const ElDeprecatedIcon = styled.span`
  display: flex;
  font-size: 1.5rem;

  &.${elIntentPrimary} {
    color: var(--intent-primary);
  }

  &.${elIntentNeutral} {
    color: var(--intent-neutral);
  }

  &.${elIntentSuccess} {
    color: var(--intent-success);
  }

  &.${elIntentPending} {
    color: var(--intent-pending);
  }

  &.${elIntentWarning} {
    color: var(--intent-warning);
  }

  &.${elIntentDanger} {
    color: var(--intent-danger);
  }

  &.${elIntentDefault} {
    color: var(--intent-default);
  }
`
