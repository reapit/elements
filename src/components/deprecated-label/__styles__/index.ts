import { styled } from '@linaria/react'

/** @deprecated This style is deprecated and will be removed in future releases. */
export const ElDeprecatedLabel = styled.label`
  font-size: var(--font-size-small);
  color: var(--neutral-500);
`
