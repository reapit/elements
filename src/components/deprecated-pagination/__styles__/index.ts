import { css } from '@linaria/core'
import { styled } from '@linaria/react'
/** @deprecated */
export const ElDeprecatedPaginationWrap = styled.div`
  width: fit-content;
  margin: 0 auto;
  display: flex;
  align-items: center;
`
/** @deprecated */
export const ElDeprecatedPaginationText = styled.div`
  margin: 0 0.5rem;
  font-size: var(--font-size-small);
`
/** @deprecated */
export const ElDeprecatedPaginationInput = styled.input`
  font-size: var(--font-size-small);
  text-align: center;
  font-weight: var(--font-weight-bold);
  margin: 0;
  font-family: var(--font-sans-serif);
  border-radius: 0.2rem;
  width: 2rem;
  height: 2rem;
  border: var(--component-input-border);

  &:focus {
    border: var(--component-input-border-focus);
  }
`
/** @deprecated */
export const ElDeprecatedPaginationButton = styled.button`
  width: 2rem;
  height: 2rem;
  border: none;
  background: var(--pagination-bg);
  border-radius: 0.2rem;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0 0.25rem;

  svg {
    color: var(--neutral-500);
    font-size: 1rem;
  }

  &:last-of-type {
    margin-right: 0;
  }
`
/** @deprecated */
export const elDeprecatedPaginationPrimary = css`
  cursor: pointer;
  svg {
    color: var(--intent-primary);
  }
`
