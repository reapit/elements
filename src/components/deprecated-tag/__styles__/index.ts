import { styled } from '@linaria/react'
import {
  elIntentPrimary,
  elIntentSuccess,
  elIntentDanger,
  elIntentWarning,
  elIntentPending,
  elIntentDefault,
  elIntentNeutral,
} from '../../../styles/intent'

/** @deprecated */
export const ElDeprecatedTag = styled.span`
  border-radius: 3rem;
  display: inline-block;
  font-size: var(--font-size-smallest);
  padding: 0.25rem 1rem;
  background: var(--neutral-400);
  color: var(--white);

  &.${elIntentPrimary} {
    background-image: linear-gradient(to right, var(--intent-primary), var(--intent-primary));
    outline-color: var(--intent-primary-dark);
  }

  &.${elIntentNeutral} {
    background-image: linear-gradient(to right, var(--intent-neutral), var(--intent-neutral));
    outline-color: var(--intent-neutral-dark);
  }

  &.${elIntentSuccess} {
    background-image: linear-gradient(to right, var(--intent-success), var(--intent-success));
    outline-color: var(--intent-success-dark);
  }

  &.${elIntentPending} {
    background-image: linear-gradient(to right, var(--intent-pending), var(--intent-pending));
    outline-color: var(--intent-pending-dark);
  }

  &.${elIntentWarning} {
    background-image: linear-gradient(to right, var(--intent-warning), var(--intent-warning));
    outline-color: var(--intent-warning-dark);
  }

  &.${elIntentDanger} {
    background-image: linear-gradient(to right, var(--intent-danger), var(--intent-danger));
    outline-color: var(--intent-danger-dark);
  }

  &.${elIntentDefault} {
    background-image: linear-gradient(to right, var(--intent-default), var(--intent-default));
    outline-color: var(--intent-default-dark);
  }
`

/** @deprecated */
export const ElDeprecatedTagGroup = styled.div`
  display: grid;
`

/** @deprecated */
export const ElDeprecatedTagGroupInner = styled.div`
  display: flex;
  flex-wrap: wrap;
  grid-auto-flow: column;
  column-gap: 0.25rem;
  row-gap: 0.25rem;
  width: fit-content;
  height: fit-content;
`
