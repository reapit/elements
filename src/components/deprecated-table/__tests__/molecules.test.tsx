import { render } from '@testing-library/react'
import {
  DeprecatedTableHeadersRow,
  DeprecatedTableHeader,
  DeprecatedTableRow,
  DeprecatedTableRowContainer,
  DeprecatedTableCell,
  TableExpandableRowTriggerCell,
  TableExpandableRow,
  TableCtaTriggerCell,
  resolveNarrowOrderClass,
  TableCellSplit,
  handleTableCtaClick,
} from '../molecules'
import * as styles from '../__styles__'

describe('DeprecatedTableHeadersRow Component', () => {
  it('should match a snapshot', () => {
    const wrapper = render(<DeprecatedTableHeadersRow />)
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})

describe('DeprecatedTableHeader Component', () => {
  it('should match a snapshot', () => {
    const wrapper = render(<DeprecatedTableHeader />)
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})

describe('DeprecatedTableRow Component', () => {
  it('should match a snapshot', () => {
    const wrapper = render(<DeprecatedTableRow />)
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})

describe('TableCellSplit Component', () => {
  it('should match a snapshot', () => {
    const wrapper = render(<TableCellSplit data="Some Data" subData="Some Longer Sub Data" />)
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})

describe('DeprecatedTableCell Component', () => {
  it('should match a snapshot with no props', () => {
    const wrapper = render(<DeprecatedTableCell />)
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should match a snapshot with full props', () => {
    const wrapper = render(
      <DeprecatedTableCell
        className="foo-bar"
        icon="add"
        darkText
        narrowLabel="Label"
        narrowIsFullWidth
        narrowOrder={1}
      />,
    )
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})

describe('TableExpandableRowTriggerCell Component', () => {
  it('should match a snapshot with no children and open', () => {
    const wrapper = render(<TableExpandableRowTriggerCell isOpen narrowIsFullWidth />)
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should match a snapshot with no children and closed', () => {
    const wrapper = render(<TableExpandableRowTriggerCell />)
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should match a snapshot with children', () => {
    const wrapper = render(
      <TableExpandableRowTriggerCell>
        <div>I am a child</div>
      </TableExpandableRowTriggerCell>,
    )
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})

describe('TableCtaTriggerCell Component', () => {
  it('should match a snapshot with no children and an icon', () => {
    const wrapper = render(<TableCtaTriggerCell icon="add" />)
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should match a snapshot with children and no icon', () => {
    const wrapper = render(
      <TableCtaTriggerCell>
        <div>I am a child</div>
      </TableCtaTriggerCell>,
    )
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should match a snapshot with no children and no icon', () => {
    const wrapper = render(<TableCtaTriggerCell />)
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})

describe('TableExpandableRow Component', () => {
  it('should match a snapshot when open', () => {
    const wrapper = render(<TableExpandableRow isOpen />)
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should match a snapshot when not open', () => {
    const wrapper = render(<TableExpandableRow isOpen />)
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})

describe('DeprecatedTableRowContainer Component', () => {
  it('should match a snapshot when open', () => {
    const wrapper = render(<DeprecatedTableRowContainer isOpen />)
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should match a snapshot when not open', () => {
    const wrapper = render(<DeprecatedTableRowContainer isOpen />)
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})

describe('resolveNarrowOrderClass', () => {
  const orders = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]

  orders.forEach((order) => {
    it(`should return the correct class for order ${order}`, () => {
      expect(resolveNarrowOrderClass(order)).toEqual(styles[`ElTableCellNarrowOrder${order}`])
    })
  })
})

describe('handleTableCtaClick', () => {
  it('should call onClick with the correct argument', () => {
    const onClick = vi.fn()

    const curried = handleTableCtaClick(onClick)

    curried()

    expect(onClick).toHaveBeenCalledTimes(1)
  })
})
