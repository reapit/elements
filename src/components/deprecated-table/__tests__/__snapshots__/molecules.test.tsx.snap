// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DeprecatedTableCell Component > should match a snapshot with full props 1`] = `
<DocumentFragment>
  <div
    class="foo-bar el-table-cell-has-dark-text el-table-narrow-cell-is-full-width el-table-cell-narrow-order1 el-deprecated-table-cell"
  >
    <span
      class="el-intent-default el-icon"
      style="font-size: 1.25rem;"
    >
      <svg
        fill="none"
        height="24"
        role="img"
        style="font-size: 1.25rem;"
        title="Icon image with name add"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 11H13V4C13 3.44772 12.5523 3 12 3C11.4477 3 11 3.44772 11 4V11H4C3.44772 11 3 11.4477 3 12C3 12.5523 3.44772 13 4 13H11V20C11 20.5523 11.4477 21 12 21C12.5523 21 13 20.5523 13 20V13H20C20.5523 13 21 12.5523 21 12C21 11.4477 20.5523 11 20 11Z"
          fill="currentColor"
        />
      </svg>
    </span>
    <div
      class="mocked-styled-10 el-deprecated-table-cell-content"
      data-narrow-label="Label"
    />
  </div>
</DocumentFragment>
`;

exports[`DeprecatedTableCell Component > should match a snapshot with no props 1`] = `
<DocumentFragment>
  <div
    class="el-table-cell-has-dark-text el-deprecated-table-cell"
  >
    <div
      class="mocked-styled-10 el-deprecated-table-cell-content"
    />
  </div>
</DocumentFragment>
`;

exports[`DeprecatedTableHeader Component > should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-1 el-deprecated-table-header"
  />
</DocumentFragment>
`;

exports[`DeprecatedTableHeadersRow Component > should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-0 el-deprecated-table-headers-row"
  />
</DocumentFragment>
`;

exports[`DeprecatedTableRow Component > should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-2 el-deprecated-table-row"
  />
</DocumentFragment>
`;

exports[`DeprecatedTableRowContainer Component > should match a snapshot when not open 1`] = `
<DocumentFragment>
  <div
    class="el-is-active el-deprecated-table-row-container"
  />
</DocumentFragment>
`;

exports[`DeprecatedTableRowContainer Component > should match a snapshot when open 1`] = `
<DocumentFragment>
  <div
    class="el-is-active el-deprecated-table-row-container"
  />
</DocumentFragment>
`;

exports[`TableCellSplit Component > should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-5 el-table-cell-split-wrap"
  >
    <div
      class="mocked-styled-6 el-table-cell-split-data"
    >
      Some Data
    </div>
    <div
      class="mocked-styled-7 el-table-cell-split-sub-data"
    >
      Some Longer Sub Data
    </div>
  </div>
</DocumentFragment>
`;

exports[`TableCtaTriggerCell Component > should match a snapshot with children and no icon 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-3 el-table-cta-cell"
    role="button"
    tabindex="0"
  >
    <div>
      I am a child
    </div>
  </div>
</DocumentFragment>
`;

exports[`TableCtaTriggerCell Component > should match a snapshot with no children and an icon 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-3 el-table-cta-cell"
    role="button"
    tabindex="0"
  >
    <div
      class="mocked-styled-11 el-table-cta-icon-container"
    >
      <span
        class="el-intent-default el-icon"
      >
        <svg
          fill="none"
          height="24"
          role="img"
          title="Icon image with name add"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 11H13V4C13 3.44772 12.5523 3 12 3C11.4477 3 11 3.44772 11 4V11H4C3.44772 11 3 11.4477 3 12C3 12.5523 3.44772 13 4 13H11V20C11 20.5523 11.4477 21 12 21C12.5523 21 13 20.5523 13 20V13H20C20.5523 13 21 12.5523 21 12C21 11.4477 20.5523 11 20 11Z"
            fill="currentColor"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TableCtaTriggerCell Component > should match a snapshot with no children and no icon 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-3 el-table-cta-cell"
    role="button"
    tabindex="0"
  />
</DocumentFragment>
`;

exports[`TableExpandableRow Component > should match a snapshot when not open 1`] = `
<DocumentFragment>
  <div
    class="el-is-active el-table-expandable-row"
  >
    <div
      class="mocked-styled-13 el-table-expandable-container"
    />
  </div>
</DocumentFragment>
`;

exports[`TableExpandableRow Component > should match a snapshot when open 1`] = `
<DocumentFragment>
  <div
    class="el-is-active el-table-expandable-row"
  >
    <div
      class="mocked-styled-13 el-table-expandable-container"
    />
  </div>
</DocumentFragment>
`;

exports[`TableExpandableRowTriggerCell Component > should match a snapshot with children 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-8 el-table-expandable-row-trigger-cell"
    role="button"
    tabindex="0"
  >
    <div>
      I am a child
    </div>
  </div>
</DocumentFragment>
`;

exports[`TableExpandableRowTriggerCell Component > should match a snapshot with no children and closed 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-8 el-table-expandable-row-trigger-cell"
    role="button"
    tabindex="0"
  >
    <span
      class="el-intent-default el-icon"
    >
      <svg
        fill="none"
        height="24"
        role="img"
        title="Icon image with name more"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 12C7 13.1046 6.10457 14 5 14C3.89543 14 3 13.1046 3 12C3 10.8954 3.89543 10 5 10C6.10457 10 7 10.8954 7 12Z"
          fill="currentColor"
        />
        <path
          d="M12 14C13.1046 14 14 13.1046 14 12C14 10.8954 13.1046 10 12 10C10.8954 10 10 10.8954 10 12C10 13.1046 10.8954 14 12 14Z"
          fill="currentColor"
        />
        <path
          d="M19 14C20.1046 14 21 13.1046 21 12C21 10.8954 20.1046 10 19 10C17.8954 10 17 10.8954 17 12C17 13.1046 17.8954 14 19 14Z"
          fill="currentColor"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;

exports[`TableExpandableRowTriggerCell Component > should match a snapshot with no children and open 1`] = `
<DocumentFragment>
  <div
    class="el-table-narrow-cell-is-full-width el-table-expandable-row-trigger-cell"
    role="button"
    tabindex="0"
  >
    <span
      class="el-intent-primary el-icon"
    >
      <svg
        fill="none"
        height="24"
        role="img"
        title="Icon image with name more"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 12C7 13.1046 6.10457 14 5 14C3.89543 14 3 13.1046 3 12C3 10.8954 3.89543 10 5 10C6.10457 10 7 10.8954 7 12Z"
          fill="currentColor"
        />
        <path
          d="M12 14C13.1046 14 14 13.1046 14 12C14 10.8954 13.1046 10 12 10C10.8954 10 10 10.8954 10 12C10 13.1046 10.8954 14 12 14Z"
          fill="currentColor"
        />
        <path
          d="M19 14C20.1046 14 21 13.1046 21 12C21 10.8954 20.1046 10 19 10C17.8954 10 17 10.8954 17 12C17 13.1046 17.8954 14 19 14Z"
          fill="currentColor"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;
